gulpfile.js

```diff
@@ -75,21 +170,41 @@ gulp.task('flow', getTask('flow'));
 gulp.task('version-check', getTask('version-check'));
 
 gulp.task('react:clean', function() {
-  return del([paths.react.lib]);
+  return del([
+    paths.react.lib,
+    paths.reactDOM.lib,
+    paths.reactNative.lib,
+  ]);
 });
 
 gulp.task('react:modules', function() {
-  return gulp
+  return merge(
+    gulp
     .src(paths.react.src)
-    .pipe(babel(babelOpts))
+    .pipe(babel(babelOptsReact))
     .pipe(flatten())
-    .pipe(gulp.dest(paths.react.lib));
+    .pipe(gulp.dest(paths.react.lib)),
+
+    gulp
+    .src(paths.reactDOM.src)
+    .pipe(babel(babelOptsReactDOM))
+    .pipe(flatten())
+    .pipe(gulp.dest(paths.reactDOM.lib)),
+
+    gulp
+    .src(paths.reactNative.src)
+    .pipe(babel(babelOptsReactNative))
+    .pipe(flatten())
+    .pipe(gulp.dest(paths.reactNative.lib))
+  );
 });
 
 gulp.task('react:extract-errors', function() {
-  return gulp
-    .src(paths.react.src)
-    .pipe(extractErrors(errorCodeOpts));
+  return merge(
+    gulp.src(paths.react.src).pipe(extractErrors(errorCodeOpts)),
+    gulp.src(paths.reactDOM.src).pipe(extractErrors(errorCodeOpts)),
+    gulp.src(paths.reactNative.src).pipe(extractErrors(errorCodeOpts))
```

Comment:
I don't think this is order dependent.

I suppose this would give duplicate errors for the files in `src/shared/**/*.js` if they had any.

Not sure how to safely merge gulp sources since those _are_ order dependent.